utils/testNLC.py

```diff
@@ -0,0 +1,178 @@
+#! /usr/bin/python
+# coding: utf-8
+
+# Copyright 2018 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Test NLC instance using utterance
+"""
+import os
+import csv
+import asyncio
+import pandas as pd
+from argparse import ArgumentParser
+import aiohttp
+
+from __init__ import UTF_8, CONFIDENCE_COLUMN, \
+    UTTERANCE_COLUMN, PREDICTED_INTENT_COLUMN, GOLDEN_INTENT_COLUMN, \
+    save_dataframe_as_csv, INTENT_JUDGE_COLUMN, \
+    TEST_OUT_FILENAME, BOOL_MAP, BASE_URL, \
+    SCORE_COLUMN
+
+test_out_header = [PREDICTED_INTENT_COLUMN, CONFIDENCE_COLUMN,
+                   SCORE_COLUMN]
+
+MAX_RETRY_LIMIT = 5
+
+
+async def post(session, json, url, sem):
+    """ Single post restrained by semaphore
+    """
+    counter = 0
+    async with sem:
+        while True:
+            try:
+                async with session.post(url, json=json, raise_for_status=True) as response:
+                    res = await response.json()
+                    return res
+            except Exception as e:
+                # Max retries reached, print out the response payload
+                if counter == MAX_RETRY_LIMIT:
+                    print(e)
+                    raise e
+                counter += 1
+                print("RETRY")
+                print(counter)
+
+
+async def fill_df(utterance, row_idx, out_df, classifier_id, apikey,
+                  iam_apikey, sem, baseUrl):
+    """ Send utterance to NLC and save response to dataframe
+    """
+    async with aiohttp.ClientSession(
+            auth=aiohttp.BasicAuth(apikey, iam_apikey)) as session:
```

Comment:
I believe Basic Auth will no longer work after the move from Cloud Foundry to Resource Groups.
In a separate pull request this month the notion of username / basic auth was removed from the rest of the WA-Testing-Tool - look at the updated `testConversation.py` in master which uses an SDK object now instead of basic auth.